pages/docs/couple-your-code/couple-your-code-time-step-sizes.md

```diff
@@ -75,6 +75,50 @@ This procedure is independent of whether a serial or a parallel coupling scheme
 For parallel coupling, both solvers run together and everything happens simultaneously in both participants, while for serial coupling, the first participant needs reach the end of the window before the second one can start.
 {% endnote %}
 
+### Possible subcycling pitfall
+
+If you are using very many small time steps in one window, you might see the following warning:
+
+```txt
+preCICE just returned a maximum time step size of <SOME DT>. Such a small value can happen if you use many sub steps per time window over multiple time windows due to added-up differences of machine precision.
+```
+
+The last time step ended up close to the end of the time-window without reaching it, leading to a very small final time step.
+Such small time steps can lead to problems in the solver.
+
+One strategy to avoid this situation is to extend the last time step of a time window preventing problematic time step sizes.
+The follow example extends the time step negotiation between the solver and preCICE to ensure the next time step size `preciceDt - solverDt` stays over some threshold `tol`.
+
+```cpp
+...
+solverDt = beginTimeStep(); // e.g. compute adaptive dt
+// Can lead to using a dt that only approximately reaches end of time window
+// dt = min(preciceDt, solverDt);
+
+// Allow some tolerance
+double tol = 10e-14;
+if (preciceDt - solverDt < tol) {
+  // Use preciceDt, if difference between preciceDt and solverDt is small enough
+  dt = preciceDt;
+} else {
+  dt = min(preciceDt, solverDt);
+}
```

Comment:
The condition already covers the case, where `preciceDt < solverDt`, so this could be simplified to the following:

```suggestion
if (preciceDt - solverDt < tol) {
  dt = preciceDt;
} else {
  dt = solverDt;
}
```

Reply:
Maybe put a comment above that this block replaced the `dt = min(preciceDt, solverDt);` known from other sections? Otherwise it's maybe a bit hard to directly see where the `dt = min(preciceDt, solverDt);` ended up. But I'm also fine with this simplification.